net/wol/src/www/widgets/widgets/wake_on_lan.widget.php

```diff
@@ -48,24 +44,24 @@
   </thead>
   <tbody>
 <?php
-    foreach ($wolcomputers as $wolent):
-      $is_active = exec("/usr/sbin/arp -an |/usr/bin/grep {$wolent['mac']}| /usr/bin/wc -l|/usr/bin/awk '{print $1;}'");?>
+    foreach ($wol->wolentry->__items as $wolent):
+      $is_active = exec("/usr/sbin/arp -an |/usr/bin/grep {$wolent->mac}| /usr/bin/wc -l|/usr/bin/awk '{print $1;}'");?>
     <tr>
-        <td><?=$wolent['descr'];?><br/><?=$wolent['mac'];?></td>
-        <td><?=htmlspecialchars(convert_friendly_interface_to_friendly_descr($wolent['interface']));?></td>
+        <td><?= $wolent->descr ?><br/><?= $wolent->mac ?></td>
+        <td><?=htmlspecialchars(convert_friendly_interface_to_friendly_descr((string)$wolent->interface));?></td>
         <td>
           <span class="glyphicon glyphicon-<?=$is_active == 1 ? "play" : "remove";?> text-<?=$is_active == 1 ? "success" : "danger";?>" ></span>
           <?=$is_active == 1 ? gettext("Online") : gettext("Offline");?>
         </td>
         <td>
-          <a href="services_wol.php?mac=<?=$wolent['mac'];?>&amp;if=<?=$wolent['interface'];?>">
+          <button class="btn btn-primary btn-sm wakeupbtn" data-mac="<?= $wolent->mac ?>" data-interface="<?= $wolent->interface ?>" data-uuid="<?= $wolent->getAttributes()['uuid'] ?>">
             <span class="glyphicon glyphicon-flash" title="<?=gettext("Wake Up");?>"></span>
-          </a>
+          </button>
         </td>
     </tr>
 <?php
     endforeach;
-    if (count($wolcomputers) == 0):?>
+    if (count($wol->wolentry->__items) == 0):?>
```

Comment:
isn't there a -&gt;count builtin?

Reply:
no, ArrayField does not have this method

Reply:
ok, thanks